tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,354 @@ def test_first_release_environments(self):
+        )
+        assert group_b_event_1.get_environment().name == u''  # has no environment
+
+        group_b = Group.objects.get(id=group_b_event_1.group.id)
+        assert group_b.id == group_b_event_1.group.id
+
+        assert group_b.first_seen == group_b.last_seen == group_b_event_1.datetime
+
+        # group_b_event_1 occurred before release_2 was created
+        assert group_b_event_1.datetime < release_2_timestamp
+
+        group_b.times_seen = 422
+        group_b.status = GroupStatus.UNRESOLVED
+        group_b.save()
+
+        # create an issue/group whose event that occur in no environments
+        # but will be tied to release_2
+
+        group_c_event_1 = self.store_event(
+            data={
+                'fingerprint': ['group_c'],
+                'event_id': 'ccc' + ('1' * 29),
+                'message': 'group_c',
+                'tags': {
+                    'server': 'example.com',
+                },
+                'timestamp': (release_2_timestamp + timedelta(days=1)).isoformat()[:19],
+                'stacktrace': {
+                    'frames': [{
+                        'module': 'group_c'
+                    }]
+                },
+            },
+            project_id=self.project.id,
+        )
+        assert group_c_event_1.get_environment().name == u''  # has no environment
+
+        group_c = Group.objects.get(id=group_c_event_1.group.id)
+        assert group_c.id == group_c_event_1.group.id
+
+        assert group_c.first_seen == group_c.last_seen == group_c_event_1.datetime
+
+        group_c.times_seen = 666
+        group_c.status = GroupStatus.UNRESOLVED
+        group_c.save()
+
+        # expect no groups since given release version does not exist
+
+        non_existent_release = 'fake'
+        assert release_1.version != non_existent_release
+        assert release_2.version != non_existent_release
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % non_existent_release,
+        )
+
+        assert set(results) == set([])
+
+        # expect no groups even though there are releases; since no group
+        # is attached to any releases
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([])
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_2.version,
+        )
+        assert set(results) == set([])
+
+        # set group_a's first_release to be release_1
+
+        group_a.first_release = release_1
+        group_a.save()
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_1.version,
+        )
+        assert set(results) == set([group_a])
+
+        results = self.make_query(
+            search_filter_query='first_release:%s' % release_2.version,
+        )
+        assert set(results) == set([])
```

Comment:
What does replaying this query help ensure? The data change above affected the query for release_1 which means the group couldn't also be in release_2.